data_structures/linked_list/linked_list.py

```diff
@@ -2,50 +2,44 @@
 
 class LinkedList:
     """class for new linked lists"""
-    def __init__(self, iter = []):
+    def __init__(self, iter=[]):
         self.head = None
-        self._len = 0 
+        self._len = 0
 
         for item in iter:
-            self.head = Node(item, self.head)
-            self._len += 1
-
-    
+            self.head = self.insert(item)
 
     def __len__(self):
         """return len of the corrent object"""
         return self._len
     
-    
     def __str__(self):
         """return all items from the LL"""
-        lis = []
+        lis = ''
         current = self.head
         for _ in range(self._len+1):
-            lis.append(current)
+            lis += str(current.val) + " "
             current = current._next
-        return str(lis) 
+        return lis
 
-    
     def insert(self, val):
         """add item to the LL"""
         self.head = Node(val, self.head)
         self._len += 1
     
-
     def find(self, val):
         """search for element and return True or false"""
-        if self.head == None:
+        if self.head is None:
             return False
-        elif self.head == val:
+        elif self.head.val == val:
             return True
         else:
```

Comment:
This can all be massively consolidated and would also fail if you were trying to find something that was not in a 1 long LL.